БИВТ-24-15 Клеймёнова Дарья blue

Dashakll opened the pull request.

Dashakll committed:
БИВТ-24-15 Клеймёнова Дарья blue

NemirovichMark commented:
Здравствуйте, Дарья.

Переименовать файлы в Blue_1 и т д
Поднять уровень доступа у классов с internal до public

Выгрузка новой системы тестирования. 
Описание ошибок можно найти в словаре:
https://newlms.misis.ru/mod/glossary/view.php?id=65847&mode=letter&hook=SPECIAL

[Lab_6_log_Blue.txt](https://github.com/user-attachments/files/19042497/Lab_6_log_Blue.txt)

Dashakll committed:
Клеймёнова БИВТ-24-15 отредактировала файлы

Dashakll commented:
Прошу прощения, не заметила файл с ошибками, чуть позже пришлю исправленные решения

NemirovichMark commented:
Окей

Dashakll committed:
БИВТ-24-15 Клеймёнова Дарья blue исправила ошибки

NemirovichMark commented:
Здравствуйте, Дарья.

Пожалуйста, запускайте работу перед запуском. И если она не запускается, значит, нужно исправить ошибки:

![image](https://github.com/user-attachments/assets/cbda7f67-e52b-4eda-be05-68a61e972e3d)
![image](https://github.com/user-attachments/assets/9643658d-aa52-4837-911e-99521238215c)
![image](https://github.com/user-attachments/assets/8a984a60-a333-44f7-b9d3-07e04530196b)

Dashakll committed:
Бивт-24-15 Клеймёнова

Добрый вечер, прошу прощения, что такие глупые ошибки оставила, с двух устройств пытаюсь работать, и vs их просто не показывал. Надеюсь, всё исправила!

NemirovichMark commented:
Здравствуйте, Дарья.

Другое дело!
Ещё поднять уровень доступа у классов с internal до public (по умолчанию, если не указан в явном виде, ставится internal)
![image](https://github.com/user-attachments/assets/52ddde6f-1e9e-42af-be59-496d0ea4ae17)

Выгрузка системы тестирования в файле. 
Описание ошибок можно найти в словаре:
https://newlms.misis.ru/mod/glossary/view.php?id=65847&mode=letter&hook=SPECIAL

[Lab_6_log_Blue.txt](https://github.com/user-attachments/files/19271803/Lab_6_log_Blue.txt)

Dashakll committed:
БИВТ-24-15 Клеймёнова blue

исправила

NemirovichMark commented:
Здравствуйте, Дарья.
Хорошо!

Во 2м номере по 1й размерности вышли за границу (2 прыжка)
for (int j = 0; j < 5; j++) { _marks[_counter, j] = result[j]; }

В 3м номере: проверка на null должна идти первой! Сперва проверяете, есть ли объект, и если да, то тогда уже его длину. А то если померить длину того, чего нет, получаем ошибку :(
 if (_pT.Length == 0 || _pT == null) return null;

В 5м номере
Тут все же  0, если спортсменов совсем нет.
 if (_sportsmen == null) return 18;

[Lab_6_log_Blue.txt](https://github.com/user-attachments/files/19273585/Lab_6_log_Blue.txt)

Dashakll commented:
Добрый вечер ещё раз, у меня вопрос на счёт вашей "подсказки" про 5 номер: переменную  _sportsmen разве можно "ретурнить" числом?
![image](https://github.com/user-attachments/assets/4012e58d-dbe0-4ff8-acba-6a393728ac23)

Dashakll committed:
БИВТ-24-15 Клеймёнова blue

Надеюсь, что верно учла Ваши замечания

NemirovichMark commented:
Здравствуйте, Дарья.

2й номер теперь в сортировке та же ошибка

3й номер
 "Лишняя" проверка при работе с массивом if (a.Length == 0) return null; - в таком случае возвращать нужно массив a. Массив нулевой длины - это не null.
 if (_pT == null || _pT.Length == 0) return null;

[Lab_6_log_Blue.txt](https://github.com/user-attachments/files/19314275/Lab_6_log_Blue.txt)

Dashakll committed:
Клеймёнова Дарья БИВТ-24-15 blue

NemirovichMark commented:
Здравствуйте, Дарья.
3й номер
"Лишняя" проверка при работе с массивом if (a.Length == 0) return null; - в таком случае возвращать нужно массив a. Массив нулевой длины - это не null.
    if (_pT == null || _pT.Length == 0) return null;

[Lab_6_log_Blue.txt](https://github.com/user-attachments/files/19428816/Lab_6_log_Blue.txt)

Dashakll committed:
Клеймёнова Дарья БИВТ-24-15

NemirovichMark commented:
Здравствуйте, Дарья.
А  public int[] PenaltyTimes?
[Lab_6_log_Blue.txt](https://github.com/user-attachments/files/19453485/Lab_6_log_Blue.txt)

Dashakll committed:
Add files via upload

NemirovichMark commented:
Здравствуйте, Дарья.

Отлично! Работа принята. Допуск к защите выдан.
Загрузите файлы на Moodle и приходите на защиту.

[Lab_6_log_Blue.txt](https://github.com/user-attachments/files/19463349/Lab_6_log_Blue.txt)
